flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/JobGraphJobInformation.java

```diff
@@ -40,11 +48,47 @@ public JobGraphJobInformation(JobGraph jobGraph) {
     }
 
     @Override
-    public VertexInformation getVertexInformation(JobVertexID jobVertexId) {
+    public JobInformation.VertexInformation getVertexInformation(JobVertexID jobVertexId) {
         return new JobVertexInformation(jobGraph.findVertexByID(jobVertexId));
     }
 
-    private static class JobVertexInformation implements VertexInformation {
+    public JobID getJobID() {
+        return jobID;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    // note: the SlotSharingGroup returned is mutable.
+    public Iterable<JobInformation.VertexInformation> getVertices() {
+        return jobGraphVerticesToVertexInformation(jobGraph.getVertices());
+    }
+
+    public static Iterable<JobInformation.VertexInformation> jobGraphVerticesToVertexInformation(
+            Iterable<JobVertex> verticesIterable) {
+        return () -> {
+            Iterator<JobVertex> iterator = verticesIterable.iterator();
+            return new Iterator<JobInformation.VertexInformation>() {
+                @Override
+                public boolean hasNext() {
+                    return iterator.hasNext();
+                }
+
+                @Override
+                public JobInformation.VertexInformation next() {
+                    return new JobVertexInformation(iterator.next());
+                }
+            };
+        };
+    }
+
+    /** Returns a copy of a jobGraph that can be mutated. */
+    public JobGraph getMutableJobGraph() throws IOException, ClassNotFoundException {
```

Comment:
I think this method name is a bit misleading. The JavaDoc say that it copies the `JobGraph`. Why not calling this method then `copyJobGraph()`?